### PR TITLE
integrate spy to vscode extension

### PR DESCRIPTION
**Pull request by mbarekh**
User can run xcomponent.preview.spy command to use the spy web

### CODE REVIEW

File: extension/package.json [hunk 1]
```diff
@@ -42,6 +42,11 @@
         "command": "xcomponent.preview.composition",
         "title": "XComponent: Preview Composition",
         "group": "navigation"
+      },
+      {
+        "command": "xcomponent.preview.spy",
```
Comment (maurelio1234) [inline extension/package.json]: it's not a preview... it should be on its own namespace... like `xcomponent.launch.spy`

Comment (maurelio1234) [inline extension/package.json]: update readme too

File: extension/src/extension.ts [hunk 2]
```diff
@@ -48,6 +53,30 @@ export function activate(context: vscode.ExtensionContext) {
...
 
+    portfinder.getPortPromise()
+        .then((port: number) => {
+            const disposableSpy = vscode.commands.registerCommand("xcomponent.preview.spy", () => {
+                const dirPath = path.parse(context.extensionPath);
+                const serverPath = `${dirPath.dir}${path.sep}spy${path.sep}server.js`;
+                const runSpyServercommand = `node ${serverPath}`;
+                (<any>process.env.port) = port;
+                const promiseCheckPortStatus = promisify(portscanner.checkPortStatus);
+                const localhost = "localhost";
```
Comment (maurelio1234) [inline extension/src/extension.ts]: why?

File: spy/bin/index.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<html>
```
Comment (maurelio1234) [inline spy/bin/index.html]: why commit a fie from bin?

File: extension/package.json (hunk 1 above)
Comment (mbarekh) [inline extension/package.json]: done

File: extension/src/extension.ts (hunk 2 above)
Comment (mbarekh) [inline extension/src/extension.ts]: done

File: spy/bin/index.html (hunk 3 above)
Comment (mbarekh) [inline spy/bin/index.html]: done

**Pull request merged**